schemas/dependencies/endpoint-monitoring-provider-1.yml

```diff
@@ -0,0 +1,79 @@
+---
+"$schema": /metaschema-1.json
+version: '1.0'
+type: object
+
+additionalProperties: false
+
+properties:
+  "$schema":
+    type: string
+    enum:
+    - /dependencies/endpoint-monitoring-provider-1.yml
+  labels:
+    "$ref": "/common-1.json#/definitions/labels"
+  name:
+    type: string
+  description:
+    type: string
+  provider:
+    type: string
+    enum:
+    - blackbox-exporter
+    description: The provider defines which monitoring implementation will be used for endpoints
+  timeout:
+    type: string
+    description: Timeout for the monitoring check, e.g. 10s
+  checkInterval:
+    type: string
+    description: Interval between endpoint checks, e.g. 10s
+  metricLabels:
+    "$ref": "/common-1.json#/definitions/labels"
+    description: Labels to be added to the resulting Prometheus metrics
+  blackboxExporter:
+      type: object
+      description: blackbox-exporter specific configuration options
+      additionalProperties: false
+      properties:
+        module:
+          type: string
+          description: the blackbox exporter module to use for the endpoint check, e.g. http_2xx
```

Comment:
Should `module` be an enum, since we know which modules there are?

Reply:
the list of modules in blackbox_exporter is not static. additional modules can be defined in, see [here](https://github.com/prometheus/blackbox_exporter#configuration). http_2xx is one of the predefined ones

Reply:
Indeed, but we control blackbox exporter, we know which modules are available. What I'm trying to avoid here is a typo in the module name not being caught at a MR level in app-interface and then making things fail much later. I prefer to maintain a list of what we accept here than facing the potential debugging of that

Reply:
i thought that things too specific for AppSRE should be avoided in schemas in the spirit of opensource.

Reply:
That's a fair point. Then the only chance we have is to add the enumeration in an app-interface setting that is checked from the integration so that we have the best of both worlds :)

Reply:
I don't want to block on this. If you agree with the solution, you can totally take care of it in a subsequent PR

Reply:
agreed. thank you for your thorough review!